testfixtures/logcapture.py

```diff
@@ -73,11 +86,53 @@ def atexit(cls):
                 '%s' % ('\n'.join((str(i.names) for i in cls.instances)))
                 )
 
+    def __len__(self):
+        return len(self.records)
+
+    def __getitem__(self, key):
+        return self._rec_2_obj(self.records[key])
```

Comment:
What is `key` here? Maybe we should use `index` for clarity? (if I understand this correctly)